drivers/bluetooth/hci/cyw43xxx.c

```diff
@@ -0,0 +1,301 @@
+
+	/* Allocate buffer for update uart baudrate command.
+	 * It will be BT_HCI_OP_RESET with extra parameters.
+	 */
+	buf = bt_hci_cmd_create(BT_HCI_VND_OP_UPDATE_BAUDRATE,
+				HCI_VSC_UPDATE_BAUD_RATE_LENGTH);
+	if (buf == NULL) {
+		LOG_ERR("Unable to allocate command buffer");
+		return -ENOMEM;
+	}
+
+	/* Add data part of packet */
+	void *data_buf = net_buf_add(buf, HCI_VSC_UPDATE_BAUD_RATE_LENGTH);
+
+	/* Copy data */
+	(void)memcpy(data_buf, (void *)hci_data, HCI_VSC_UPDATE_BAUD_RATE_LENGTH);
+
+	/* Send update uart baudrate command. */
+	err = bt_hci_cmd_send_sync(BT_HCI_VND_OP_UPDATE_BAUDRATE, buf, NULL);
+	if (err) {
+		return err;
+	}
+
+	/* Re-configure Uart baudrate */
+	err = bt_hci_uart_set_baudrate(bt_uart_dev, baudrate);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int bt_firmware_download(const uint8_t *firmware_image, uint32_t size)
+{
+	uint8_t *data = (uint8_t *)firmware_image;
+	volatile uint32_t remaining_length = size;
+	struct net_buf *buf;
+	int err;
+
+	LOG_DBG("Executing Fw downloading for CYW43xx device");
+
+	/* Send hci_download_minidriver command */
+	err = bt_hci_cmd_send_sync(BT_HCI_VND_OP_DOWNLOAD_MINIDRIVER, NULL, NULL);
+	if (err) {
+		return err;
+	}
+
+	/* The firmware image (.hcd format) contains a collection of hci_write_ram
+	 * command + a block of the image, followed by a hci_write_ram image at the end.
+	 * Parse and send each individual command and wait for the response. This is to
+	 * ensure the integrity of the firmware image sent to the bluetooth chip.
+	 */
+	while (remaining_length) {
+		size_t data_length = data[2]; /* data length from firmware image block */
+		uint16_t op_code = *(uint16_t *)data;
+
+		/* Allocate buffer for hci_write_ram/hci_launch_ram command. */
+		buf = bt_hci_cmd_create(op_code, data_length);
+		if (buf == NULL) {
+			LOG_ERR("Unable to allocate command buffer");
+			return err;
+		}
+
+		/* Add data part of packet */
+		void *data_buf = net_buf_add(buf, data_length);
+
+		(void)memcpy(data_buf, (void *)&data[3], data_length);
+
+		/* Send hci_write_ram command. */
+		err = bt_hci_cmd_send_sync(op_code, buf, NULL);
+		if (err) {
+			return err;
+		}
+
+		switch (op_code) {
+		case BT_HCI_VND_OP_WRITE_RAM:
+			/* Update remaining length and data pointer:
+			 * content of data length + 2 bytes of opcode and 1 byte of data length.
+			 */
+			data += data_length + 3;
+			remaining_length -= data_length + 3;
+			break;
+
+		case BT_HCI_VND_OP_LAUNCH_RAM:
+			remaining_length = 0;
+			break;
+
+		default:
+			return -ENOMEM;
+		}
+	}
+
+	LOG_DBG("Fw downloading complete");
+	return 0;
+}
+
+int bt_h4_vnd_setup(const struct device *dev)
+{
+	int err;
+	uint32_t default_uart_speed = DT_PROP(DT_CHOSEN(zephyr_bt_uart), current_speed);
+	uint32_t baudrate_for_hci_operation =
+		DT_INST_PROP_OR(0, hci_operation_speed, default_uart_speed);
+	uint32_t baudrate_for_fw_download =
+		DT_INST_PROP_OR(0, fw_download_speed, default_uart_speed);
```

Comment:
```suggestion
	uint32_t hci_operation_speed =
		DT_INST_PROP_OR(0, hci_operation_speed, default_uart_speed);
	uint32_t fw_download_speed =
		DT_INST_PROP_OR(0, fw_download_speed, default_uart_speed);
```
Nit: I would rename variables to follow devicetree properties. That way it should be easier to follow this function logic with used devicetree.